seq2seq/nmt_without_attention_v2.py

```diff
@@ -0,0 +1,243 @@
+            if event.batch_id % 100 == 0 and event.batch_id > 0:
+                with gzip.open('models/nmt_without_att_params_batch_%d.tar.gz' %
+                               event.batch_id, 'w') as f:
+                    parameters.to_tar(f)
+
+            if event.batch_id % 10 == 0:
+                #                wmt14_test_batch = paddle.batch(
+                #                        paddle.reader.shuffle(
+                #                            paddle.dataset.wmt14.test(source_dict_dim), 
+                #                            buf_size=8192), batch_size=1)
+
+                #test_result = trainer.test(wmt14_test_batch)
+                print "\nPass %d, Batch %d, Cost %f, %s" % (
+                    event.pass_id,
+                    event.batch_id,
+                    event.cost,
+                    event.metrics,  # test_result.cost, test_result.metrics
+                )
+            else:
+                sys.stdout.write('.')
+                sys.stdout.flush()
+
+    # start to train
+    trainer.train(
+        reader=wmt14_reader, event_handler=event_handler, num_passes=2)
+
+
+def generate(source_dict_dim, target_dict_dim):
+    # use the first 3 samples for generation
+    gen_creator = paddle.dataset.wmt14.gen(source_dict_dim)
+    gen_data = []
+    gen_num = 3
+    for item in gen_creator():
+        gen_data.append((item[0], ))
+        if len(gen_data) == gen_num:
+            break
+
+    beam_gen = seq2seq_net(source_dict_dim, target_dict_dim, True)
+    # get the pretrained model, whose bleu = 26.92
+    # parameters = paddle.dataset.wmt14.model()
+    with gzip.open('models/nmt_without_att_params_batch_400.tar.gz') as f:
+        parameters = paddle.parameters.Parameters.from_tar(f)
+    # prob is the prediction probabilities, and id is the prediction word. 
+    beam_result = paddle.infer(
+        output_layer=beam_gen,
+        parameters=parameters,
+        input=gen_data,
+        field=['prob', 'id'])
+
+    # get the dictionary
+    src_dict, trg_dict = paddle.dataset.wmt14.get_dict(source_dict_dim)
+
+    # the delimited element of generated sequences is -1,
+    # the first element of each generated sequence is the sequence length
+    seq_list = []
+    seq = []
+    for w in beam_result[1]:
+        if w != -1:
+            seq.append(w)
+        else:
+            seq_list.append(' '.join([trg_dict.get(w) for w in seq[1:]]))
+            seq = []
+
+    prob = beam_result[0]
+    beam_size = 3
+    for i in xrange(gen_num):
+        print "\n*******************************************************\n"
+        print "src:", ' '.join([src_dict.get(w) for w in gen_data[i][0]]), "\n"
+        for j in xrange(beam_size):
+            print "prob = %f:" % (prob[i][j]), seq_list[i * beam_size + j]
+
+
+def main():
+    paddle.init(use_gpu=False, trainer_count=4)
+    source_language_dict_dim = 30000
+    target_language_dict_dim = 30000
+    generating = True
```

Comment:
please make `generating` a parameter and get it from the command line. Now is set to `True`, if a user directly runs this example, it is not logic because we do not provide a trained model, he cannot generate without training.